internal/app_config.go

```diff
@@ -0,0 +1,127 @@
+package internal
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/pelletier/go-toml"
+)
+
+var DEF_CONFIG_ORDER = []string{
+	"/usr/lib/sota/conf.d",
+	"/var/sota/sota.toml",
+	"/etc/sota/conf.d/",
+}
+
+type cfgFile struct {
+	name string
+	path string
+
+	tree *toml.Tree
+}
+
+type AppConfig struct {
+	cfgs []*cfgFile
+}
+
+// NewAppConfig parses config files as per:
+// https://docs.ota.here.com/ota-client/latest/aktualizr-config-options.html#_how_toml_files_are_processed
+func NewAppConfig(configPaths []string) (*AppConfig, error) {
+	configsMap := make(map[string]string)
+
+	// Build up the files
+	for _, path := range configPaths {
+		if st, err := os.Stat(path); err == nil {
+			if st.IsDir() {
+				if entries, err := os.ReadDir(path); err == nil {
+					for _, entry := range entries {
+						if strings.HasSuffix(entry.Name(), ".toml") && entry.Type().IsRegular() {
+							configsMap[entry.Name()] = filepath.Join(path, entry.Name())
+						}
+					}
+				}
+			} else {
```

Comment:
Not sure about it, but maybe we should skip links?

Reply:
libaktualizr doesn't seem to:
https://github.com/uptane/aktualizr/blob/master/src/libaktualizr/config/base_config.cc#L27

Reply:
I'm on the fence here.

My original comment was going to read like this:
> I'd check for `IsRegular` here.
> Every month there are a plenty of new CVEs about privilege escalation vulnerabilities found in software following symlinks.
> So, unless we have a _strong reason_ to support symlinks here, I'd not support them.

But then I thought: hey, this process is needs access to system files, so what privilege escalation is possible.

So, I'd leave it up to you here.

But, bear in mind, that if you want to support symlinks here - then you need to also support symlinks at line 41 for consistency.

Reply:
Its not so much I want to support symlinks, but if we are going to change this - then we also need to change libaktuzlizr

Reply:
If you don't want to change this - then you need to change line 41 above; as the symlink is not supported out there.

Reply:
https://github.com/foundriesio/fioconfig/pull/45/commits/348421ef08fb4d9e062b06813387d0b565d19e32